### PR TITLE
Add onWelcomePageOpen event

### PR DESCRIPTION
**Pull request by chlafreniere**
We've had a lot of discussions around the best way to handle commands that come from extensions that are listed on the welcome page. Here's a proposal to add an activation event that extensions can hook into for when the welcome page is open. Please let me know what you think 😄

### CODE REVIEW

File: src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 1]
```diff
@@ -346,6 +348,8 @@ class WelcomePage extends Disposable {
 				}
 			}
 		}));
+		// {{SQL CARBON EDIT}} - Add onWelcomePageOpen event
```
Comment (Charles-Gagnon) [inline src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: Put this on the same line as the added line

Comment (chlafreniere) [reply src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: Thanks done

**Comment by anthonydresser** [comment]
Can we have an in person discussion on this. I'm still not sure this solves the issue.

**Comment by chlafreniere** [comment]
This approach doesn't solve the problem sufficiently; there is still one scenario where this doesn't work as intended and clicking a command link does nothing. I'll focus on the other PR on this topic.

**Pull request closed without merging**